spark-connector/spark-common/src/main/java/org/apache/gravitino/spark/connector/hive/SparkHiveTypeConverter.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.gravitino.spark.connector.hive;
+
+import org.apache.gravitino.rel.types.Type;
+import org.apache.gravitino.rel.types.Types;
+import org.apache.gravitino.spark.connector.SparkTypeConverter;
+import org.apache.spark.sql.types.DataType;
+import org.apache.spark.sql.types.DataTypes;
+
+public class SparkHiveTypeConverter extends SparkTypeConverter {
+  @Override
+  public DataType toSparkType(Type gravitinoType) {
+    if (gravitinoType instanceof Types.TimestampType) {
+      return DataTypes.TimestampType;
```

Comment:
According to the logic all Gravitino timestamp types are converted Spark `DataTypes.TimestampType`,  but why the expected column are  `DataTypes.TimestampNTZType` in Spark34 and Spark35 ? do I miss something?

Reply:
Should you also consider modifying the code here to throw an exception, in order to align its behavior with that of the Spark Hive connector?
https://github.com/apache/gravitino/blob/3f648ad07d721b1b932154f85443429282bd4c87/spark-connector/spark-common/src/main/java/org/apache/gravitino/spark/connector/SparkTypeConverter.java#L82-L83

Reply:
> Should you also consider modifying the code here to throw an exception, in order to align its behavior with that of the Spark Hive connector?
> 
> https://github.com/apache/gravitino/blob/3f648ad07d721b1b932154f85443429282bd4c87/spark-connector/spark-common/src/main/java/org/apache/gravitino/spark/connector/SparkTypeConverter.java#L82-L83

@mchades There should be no need for processing here. It is common to all types of Catalogs, and special handling is required for Hive.

Reply:
> If Spark3.3 use `TimestampType` while Spark3.4 and Spark 3.5 use `TimestampNTZType` to represent Hive `timestamp`, the query result will different if using different Spark version to query same hive table, Yes?

@FANNG1 My consideration is that for Spark 3.3, users can consider performing a display conversion.

Reply:
> Should you also consider modifying the code here to throw an exception, in order to align its behavior with that of the Spark Hive connector?
> 
> https://github.com/apache/gravitino/blob/3f648ad07d721b1b932154f85443429282bd4c87/spark-connector/spark-common/src/main/java/org/apache/gravitino/spark/connector/SparkTypeConverter.java#L82-L83

@mchades There should be no need for processing here. It is common to all types of Catalogs, and special handling is required for Hive.

Reply:
But I didn't see you rewrite this method in this PR.

Reply:
Yes, I haven't made any changes yet because this PR is based on the implementation of timestamp compatibility with spark33. If it is confirmed that timestamp support is not needed, I will continue to make changes. But it still feels a bit strange to do so.

Reply:
> this PR is based on the implementation of timestamp compatibility with spark33.

do you mean there will be another PR to handle spark3.4 and spark3.5?